### ISSUE TITLE
Tax rouding up instead of down.

### ISSUE THREAD

**Issue by pmonosolo** [magento/magento2#35025]
### Preconditions (*)

1. Magento 2.4.3 CE
2. Product with price $38.48
3. Taxes: Canadian GST (5%) and PST(7%) taxes applied to order
4. GST tax: 38.48 * 0.5 = 1.924 ~ 1.93 
5. PST tax $38.48 * 0.7 =  2.6936 ~ 2.7

![image](https://user-images.githubusercontent.com/66754347/151220537-8427088f-e5f2-481a-98b8-b819b4e16753.png)

Both PST and GST are being rounded up even though they should not be. I believe it should be rounding up only at 0.5 not at 0.4.

This creates a misalignment when orders get synced into Quickbooks. 

![image](https://user-images.githubusercontent.com/66754347/151226933-3e4a9272-164d-4f73-9132-a0afe8910929.png)

### Steps to reproduce (*)

1. Create a product with price $38.48

2. Create Tax Rate for GST

![image](https://user-images.githubusercontent.com/66754347/151221057-a5812a07-5e72-4c33-ad96-a0af1e78c7e0.png)

3. Create Tax Rate for PST

![image](https://user-images.githubusercontent.com/66754347/151221145-67261479-d892-4cb9-bc3e-09571b19e5d2.png)

4. Create Tax Rule for for GST

![image](https://user-images.githubusercontent.com/66754347/151221256-239c5e87-ffaa-4f9d-91fb-0ddb0b90e5f7.png)

5. Create Tax Rule for PST

![image](https://user-images.githubusercontent.com/66754347/151221346-3257fb39-45cd-476c-963f-ec1466b6042b.png)

6. Put product with price 38.48 into cart and go to checkout

7. Check out with address from Canada, British Columbia

### Expected result (*)

1. Order Total should be $43.09

### Actual result (*)

1. Order Total Shows $43.10 (which actually is incorrect even by the broken rouding up system - 38.48 + 2.70 + 1.93 = 43.11)

![image](https://user-images.githubusercontent.com/66754347/151221597-9eeedb26-fe43-4daa-850e-97f7916c76e7.png)

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._

**Comment by pmonosolo** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-November** [comment]
Hi @pmonosolo,

Thank you for reporting the issue.

I have tried this issue on the fresh installation (Magento 2.4-dev/2.4.3-P1) but I am not able to reproduce it with given steps
![image](https://user-images.githubusercontent.com/60198231/151378598-91443c68-b28b-4f1c-b3ac-ef067ea6e4bc.png)

Kindly recheck the issue on Magento 2.4-develop branch and elaborate the steps if the issue is still reproducible.
Thank you.

**Comment by pmonosolo** [comment]
> Hi @pmonosolo,
> 
> Thank you for reporting the issue.
> 
> I have tried this issue on the fresh installation (Magento 2.4-dev/2.4.3-P1) but I am not able to reproduce it with given steps ![image](https://user-images.githubusercontent.com/60198231/151378598-91443c68-b28b-4f1c-b3ac-ef067ea6e4bc.png)
> 
> Kindly recheck the issue on Magento 2.4-develop branch and elaborate the steps if the issue is still reproducible. Thank you.

What you have in your screenshot is exactly the issue:

![image](https://user-images.githubusercontent.com/66754347/151407789-ce6c10e8-46be-42ca-a9ee-4f23cfe328af.png)

This is how Magento Counts it:
GST tax: 38.48 * 0.05 = 1.924 ~ 1.93
PST tax $38.48 * 0.07 = 2.6936 ~ 2.7

Total Amount: $4.10  - WRONG

What it should be is:
GST tax: 38.48 * 0.05 = 1.924 ~ 1.92
PST tax $38.48 * 0.07 = 2.6936 ~ 2.69

Total Amount: $4.09 - CORRECT

**Comment by engcom-November** [comment]
Hi @pmonosolo,
Thank you for confirmation. Verified the issue again on Magento 2.4-develop branch and the issue is reproducible:
**Issue 1:** GST Tax: 5 % of (38.48) = 1.924. **Rounding up 1.924 should be 1.92 but not 1.93**
**Issue 2:** Total Amount value displayed on UI is 43.10. But Breakup details displayed are: 38.48 + 1.93 + 2.70 = 43.11. Breakup details summation is not matching with Total Amount displayed value.

**Expected Result:**
GST Tax 1.924 cannot not be rounded of to 1.93
Payment Breakup details summation should match with Total amount.

**Comment by pmonosolo** [comment]
> Hi @pmonosolo, Thank you for confirmation. Verified the issue again on Magento 2.4-develop branch and the issue is reproducible: **Issue 1:** GST Tax: 5 % of (38.48) = 1.924. **Rounding up 1.924 should be 1.92 but not 1.93** 

Correct, $1.924 should be rounded down, not up. So it should be 1.92 instead of 1.93.

From what I understand the rounding happens when all of the taxes are added up.

So $1.924 + $2.6936 = $4.6176 (rounds up) = $4.62

Total amount: $38.48 + $4.62 = $43.10

This is why this Total comes out.

The taxes should be rounded out on the TAX ROW level. 

So each tax is rounded (up / down)  based on its own row. 

GST: $1.924  ~ $1.92
PST: $2.6936 ~ $2.69

Total Tax: $1.92 + $2.69 = $4.61
Total amount: $38.48 + $4.61 = $43.09

There should be at least an option for this type of calculation if we don't want to dramatically change everybody's tax calculations.

Quickbooks software calculates these taxes based on each ROW item, so I would assume that is a correct approach. 

I have tried to go through the Tax calculation methods(Total / Row Total / Unit Price):

![image](https://user-images.githubusercontent.com/66754347/151598804-cf7be50e-3b94-4339-8bd8-299eeea41410.png)

But, nothing changed in terms of the calculation.

> **Issue 2:** Total Amount value displayed on UI is 43.10. But Breakup details displayed are: 38.48 + 1.93 + 2.70 = 43.11. Breakup details summation is not matching with Total Amount displayed value.
> 
> **Expected Result:** GST Tax 1.924 cannot not be rounded of to 1.93 Payment Breakup details summation should match with Total amount.

Yes this is another weird bug, the numbers do not match the total. This fix is probably something to look after the bug 1) is fixed.

**Comment by pmonosolo** [comment]
I'm digging into this issue a bit more - I'm getting really weird numbers.

`$this->orderItemTaxFactory->create()->getTaxItemsByOrderId($orderId);`

^^^ this shows numbers that do not make any sense:

`[{"tax_id":"338","tax_percent":"7.0000","item_id":"235","taxable_item_type":"product","associated_item_id":null,"real_amount":"2.6950","real_base_amount":"2.6950","code":"PST-CA-BC","title":"PST-CA-BC","order_id":"229"},{"tax_id":"339","tax_percent":"5.0000","item_id":"235","taxable_item_type":"product","associated_item_id":null,"real_amount":"1.9250","real_base_amount":"1.9250","code":"GST-CA-BC","title":"GST-CA-BC","order_id":"229"}]`

GST
"real_amount":"1.9250"
"real_base_amount":"1.9250"

PST
"real_amount":"2.6950"
"real_base_amount":"2.6950"

How are these numbers possible if maximum number that is achievable is:

GST: 1.924
PST: 2.6936

You can't round up to these values even.

**Comment by pmonosolo** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Delta** [comment]
Hi @pmonosolo ,

Thanks for your reporting and collaboration.
We have re-verified the issue in latest 2.4-develop instance and the issue is reproducible.
Kindly refer the screenshots.

**Steps to reproduce**
1. Product with price $38.48
2. Taxes: Canadian GST (5%) and PST(7%) taxes applied to order
3. GST tax: 38.48 * 0.5 = 1.924 ~ 1.93
4. PST tax $38.48 * 0.7 = 2.6936 ~ 2.7
5. Total tax should be 1.92 + 2.69 = 4.61
![image](https://github.com/user-attachments/assets/1058652a-56fd-4801-bae8-e7db825f02d0)

Thanks.

**Comment by engcom-Bravo** [comment]
Hi @pmonosolo,

Thanks for your reporting and collaboration.

We have tried to reproduce the issue in Latest 2.4-develop instance and still we are able to reproduce the issue.Please refer the screenshot.


<img width="458" height="614" alt="Image" src="https://github.com/user-attachments/assets/6ec746ee-547f-4a86-977f-4b091b474e90" />



Hence Confirming the issue.

Thanks.
